desk/app/activity.hoon

```diff
@@ -71,59 +71,192 @@
-::     ?~  entry  next
+++  groups-prefix  /(scot %p our.bowl)/groups/(scot %da now.bowl)
+++  channels-prefix  /(scot %p our.bowl)/channels/(scot %da now.bowl)
+++  set-channel-reads
+  |=  =channels:c
+  ^+  cor
+  =+  .^(=unreads:c %gx (welp channels-prefix /v1/unreads/noun))
+  =/  entries  ~(tap by unreads)
+  =;  events=(list [time incoming-event:a])
+    |-
+    ?~  events  cor
+    =.  cor  (%*(. add should-notify |, start-time -.i.events) +.i.events)
+    $(events t.events)
+  |-  ^-  (list [time incoming-event:a])
+  ?~  entries  ~
+  =/  head  i.entries
+  =*  next  $(entries t.entries)
+  =/  [=nest:c =unread:c]  head
+  =/  channel  (~(get by channels) nest)
+  ?~  channel  next
+  =/  group  group.perm.u.channel
+  =;  events=(list [time incoming-event:a])
+    (weld events next)
+  =/  posts=(list [time incoming-event:a])
+    ?~  unread.unread  ~
+    %+  murn
+      (tab:on-posts:c posts.u.channel `(sub id.u.unread.unread 1) count.u.unread.unread)
+    |=  [=time post=(unit post:c)]
+    ?~  post  ~
+    =/  key=message-key:a
+      :_  time
+      [author.u.post sent.u.post]
+    =/  mention
+      (was-mentioned:ch-utils content.u.post our.bowl)
+    `[time %post key nest group content.u.post mention]
+  =/  replies=(list [time incoming-event:a])
+    %-  zing
+    %+  murn
+      ~(tap by threads.unread)
+    |=  [=id-post:c [id=id-reply:c count=@ud]]
+    ^-  (unit (list [time incoming-event:a]))
+    =/  post=(unit (unit post:c))  (~(get by posts.u.channel) id-post)
+    ?~  post  ~
+    ?~  u.post  ~
+    %-  some
+    %+  turn
+      (tab:on-replies:c replies.u.u.post `(sub id 1) count)
+    |=  [=time =reply:c]
+    =/  key=message-key:a
+      :_  time
+      [author.reply sent.reply]
+    =/  parent=message-key:a
+      :_  id-post
+      [author.u.u.post sent.u.u.post]
+    =/  mention
+      (was-mentioned:ch-utils content.reply our.bowl)
+    [time %reply key parent nest group content.reply mention]
+  =/  init-time
+    ?:  &(=(posts ~) =(replies ~))  recency.unread
+    *@da
+  :-  [init-time %chan-init nest group]
+  (welp posts replies)
+++  chat-prefix  /(scot %p our.bowl)/chat/(scot %da now.bowl)
+++  set-chat-reads
+  ^+  cor
+  =+  .^(=unreads:ch %gx (welp chat-prefix /unreads/noun))
+  =+  .^  [dms=(map ship dm:ch) clubs=(map id:club:ch club:ch)]
+      %gx  (welp chat-prefix /full/noun)
+    ==
+  =/  entries  ~(tap by unreads)
+  =;  events=(list [time incoming-event:a])
+    |-
+    ?~  events  cor
+    =.  cor  (%*(. add should-notify |, start-time -.i.events) +.i.events)
+    $(events t.events)
+  |-  ^-  (list [time incoming-event:a])
+  ?~  entries  ~
+  =/  head  i.entries
+  =*  next  $(entries t.entries)
+  =/  [=whom:ch =unread:unreads:ch]  head
+  =/  =pact:ch
+    ?-  -.whom
+      %ship  pact:(~(gut by dms) p.whom *dm:ch)
+      %club  pact:(~(gut by clubs) p.whom *club:ch)
+    ==
+  =;  events=(list [time incoming-event:a])
+    (weld events next)
+  :-  [*@da %dm-invite whom]
```

Comment:
We want to use the `recency.unread` here too, like in the channels case, if we didn't add any messages.

Reply:
But apparently even in the channels case it's bugged and doesn't affect the ordering as expected. We should investigate & fix prior to release.